### DISCUSSION TITLE
Personal assistant for online communities

### DISCUSSION THREAD

**Discussion by erlend-sh** [khoj-ai/khoj#380] [Ideas]
Seems like Khoj could work really well as an opt-in assistant on a messaging board like Discord or Discourse:

https://blog.discourse.org/2023/04/introducing-discourse-ai/
https://meta.discourse.org/t/discourse-ai-self-hosted-guide/259598

A forum is a type of hive-mind. "An AI personal assistant for your digital **hive**-brain" can't be that different from the use case of an individual brain.

I'm specifically interested in exploring the use of Khoj in [Commune](https://blog.erlend.sh/communal-bonfires), a digital-gardening extension of Matrix.

**Comment by erlend-sh** [comment]
Does your stated support for 'github repositories' include Issues and Discussions? That's most of the forum API covered already.

**Reply by debanjum** [reply]
The Khoj Github plugin supports indexing Issues, not Discussions (yet). See the [Khoj FAQ](https://faq.khoj.dev) for an example. 

Could you elaborate on how indexing Github Issues and Discussions could suffice for your online communties use-case? If this is all that is required, we could extend Khoj to:
1. Index discussions in a Github Repository
2. Allow configuring which of _commits_, _documents_, _issues_ and _discussions_ in a Github repository should be indexed by Khoj
